lib/skate/detours/route_segments.ex

```diff
@@ -93,24 +93,28 @@ defmodule Skate.Detours.RouteSegments do
         connection_start,
         connection_end
       ) do
-    {nearest_start_point, start_index} =
-      NearestPoint.nearest_point_on_shape(shape, connection_start)
-
-    {nearest_end_point, end_index} =
-      NearestPoint.nearest_point_on_shape(shape, connection_end)
+    {before_start_point, after_start_point} = split_shape_at_point(shape, connection_start)
+    {between_points, after_end_point} = split_shape_at_point(after_start_point, connection_end)
 
     {:ok,
      %__MODULE__.Result{
-       before_detour:
-         Enum.slice(shape, 0..start_index) ++
-           [nearest_start_point],
-       detour:
-         [nearest_start_point] ++
-           Enum.slice(shape, (start_index + 1)..end_index) ++
-           [nearest_end_point],
-       after_detour:
-         [nearest_end_point] ++
-           Enum.slice(shape, (end_index + 1)..-1)
+       before_detour: before_start_point,
+       detour: between_points,
+       after_detour: after_end_point
      }}
   end
+
+  @spec split_shape_at_point(nonempty_list(Util.Location.From.t()), Util.Location.From.t()) :: {
```

Comment:
This feels like it might go away once I address [this comment](https://github.com/mbta/skate/pull/2640#discussion_r1642749316) from an earlier PR, but as it stands, I still think this is an improvement over what it replaces.